### PR TITLE
ci: test on kubernetes 1.22

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

1.22 just got released. We now bump our tests one version higher too.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: .github/workflows/integration-test-helm-suite.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        kubernetes-versions : ['v1.15.12', 'v1.21.0']
+        kubernetes-versions : ['v1.16.15', 'v1.22.0']
```
Comment (subhamkrai) [inline .github/workflows/integration-test-helm-suite.yaml:20]:
```suggestion
        kubernetes-versions : ['v1.15.12', 'v1.22.0']
```
same multiple places where `v1.15.12` has been changed

**Review by subhamkrai — CHANGES_REQUESTED** [review]
due to v1beta resources, we want to keep k8s 1.115.x for testing

**Comment by travisn** [comment]
> due to v1beta resources, we want to keep k8s 1.115.x for testing

@subhamkrai @leseb Since we are anyway bumping the minimum version to 1.16 in v1.8, what about the following?
- With this PR we go ahead and bump the min version tested to 1.16 and add 1.22 to the test matrix. We would keep these changes only in master.
- Prior to this PR merging, are there any other github actions that need to be backported to release-1.7? I don't think so, but just trying to avoid future merge conflicts.
- Prior to this PR merging, we backport all the new actions to release-1.6? Then we can base future 1.6 patch releases on the actions and consider turning off Jenkins soon.

**Comment by subhamkrai** [comment]
> > due to v1beta resources, we want to keep k8s 1.115.x for testing
> 
> @subhamkrai @leseb Since we are anyway bumping the minimum version to 1.16 in v1.8, what about the following?
> 
>     * With this PR we go ahead and bump the min version tested to 1.16 and add 1.22 to the test matrix. We would keep these changes only in master.
> 
>     * Prior to this PR merging, are there any other github actions that need to be backported to release-1.7? I don't think so, but just trying to avoid future merge conflicts.
> 
>     * Prior to this PR merging, we backport all the new actions to release-1.6? Then we can base future 1.6 patch releases on the actions and consider turning off Jenkins soon.

right, we are going to support min k8s v1.16 in rook v1.8. Sounds good with backport approach.

**Comment by leseb** [comment]
> > due to v1beta resources, we want to keep k8s 1.115.x for testing
> 
> @subhamkrai @leseb Since we are anyway bumping the minimum version to 1.16 in v1.8, what about the following?
> 
> * With this PR we go ahead and bump the min version tested to 1.16 and add 1.22 to the test matrix. We would keep these changes only in master.
> * Prior to this PR merging, are there any other github actions that need to be backported to release-1.7? I don't think so, but just trying to avoid future merge conflicts.
> * Prior to this PR merging, we backport all the new actions to release-1.6? Then we can base future 1.6 patch releases on the actions and consider turning off Jenkins soon.

So I have no code change left, the PR is good as-is if I read you correctly.

**Review by subhamkrai — APPROVED** [review]

**Review by travisn — APPROVED** [review]

**Pull request merged**